namesys/ipns_publisher.go

```diff
@@ -189,47 +188,45 @@ func (p *IpnsPublisher) updateRecord(ctx context.Context, k crypto.PrivKey, valu
 	}
 
 	// Put the new record.
-	key := IpnsDsKey(id)
-	if err := p.ds.Put(ctx, key, data); err != nil {
+	dsKey := IpnsDsKey(id)
+	if err := p.ds.Put(ctx, dsKey, data); err != nil {
 		return nil, err
 	}
-	if err := p.ds.Sync(ctx, key); err != nil {
+	if err := p.ds.Sync(ctx, dsKey); err != nil {
 		return nil, err
 	}
+
 	return r, nil
 }
 
-// PutRecordToRouting publishes the given entry using the provided ValueStore,
-// keyed on the ID associated with the provided public key. The public key is
-// also made available to the routing system so that entries can be verified.
-func PutRecordToRouting(ctx context.Context, r routing.ValueStore, k crypto.PubKey, rec *ipns.Record) error {
-	ctx, span := StartSpan(ctx, "PutRecordToRouting")
+// PublishIPNSRecord publishes the given [ipns.Record] for the provided [crypto.PubKey] in
+// the provided [routing.ValueStore]. The public key is also made available to the routing
+// system if it cannot be derived from the corresponding [peer.ID].
+func PublishIPNSRecord(ctx context.Context, r routing.ValueStore, pubKey crypto.PubKey, rec *ipns.Record) error {
+	ctx, span := startSpan(ctx, "PutRecordToRouting")
```

Comment:
:thought_balloon:  @hacdias  you renamed funcs but the span name is the same.
Do we need to rename funcs? Maybe keep the old func name for now? (Or at least keep them in sync).
(here and in other cases like this)

Reply:
Btw, can you also look at https://github.com/ipfs/boxo/pull/459#discussion_r1315162250 and https://github.com/ipfs/boxo/pull/459#discussion_r1315162377?

Reply:
I have pushed a commit with the fix.